### PR TITLE
Starting each analysis from scratch in end2end tests.

### PR DESCRIPTION
**Pull request by isoos**
As a follow-up to #1346, this makes sure that all end2end test start with a fresh temp directory, pub cache and configuration, making sure that one temporary error at one analysis does not hide an issue with a later analysis.

### CODE REVIEW

File: test/end2end_test.dart [hunk 1]
```diff
@@ -18,56 +18,47 @@ final _goldenDir = p.join('test', 'goldens', 'end2end');
...
   }) {
     final filename = '$package-$version.json';
     group('end2end: $package $version', () {
+      late String tempDir;
+      late PackageAnalyzer analyzer;
+      http.Client? httpClient;
+      late HttpServer httpServer;
       late final Map<String, Object?> actualMap;
 
       setUpAll(() async {
```
Comment (sigurdm) [inline test/end2end_test.dart:34]: Should all this setup code not be in a `setUp` rather than `setUpAll`?

Comment (isoos) [reply test/end2end_test.dart:34]: We have multiple `test()` methods, I'd rather keep it like that.

Comment (sigurdm) [reply test/end2end_test.dart:34]: But doesn't that still create only one analysis context for all tests?

Comment (isoos) [reply test/end2end_test.dart:34]: `setupAll` runs only a single time for all the tests in the group, I think that's what we want here. Not sure: what do you see as a problem here?

Comment (sigurdm) [reply test/end2end_test.dart:34]: Ah - I think I saw it as residing at a higher level - sorry.

**Review by sigurdm — APPROVED** [review]

**Pull request merged**